### PR TITLE
Mcs 909 write target visibility

### PR DESCRIPTION
**Pull request by cedorman**
For one of the scorecard values, we need to know whether or not the target is visible.  Add this to the step metadata output.  This means a change to the step metadata object, _init__, and __iter__, and associated unit test.   

The determination of whether or not the target is visible is done in history_writer.  It's not clear that this is the right place for the logic, but there are other manipulations for of the step metadata and it was the easiest place to put it.

### CODE REVIEW

**Comment by deanwetherby** [comment]
How about moving the visibility flag to SceneHistory? It's still ingested and we wouldn't have the serialization issues.

File: machine_common_sense/history_writer.py [hunk 1]
```diff
@@ -162,13 +162,31 @@ def add_step(self, step_obj: Dict):
         """Add a new step to the array of history steps"""
         current_time = perf_counter() * 1000
         if step_obj is not None:
-            step_obj.delta_time_millis = current_time - \
-                self.last_step_time_millis
+            step_obj.delta_time_millis = \
+                current_time - self.last_step_time_millis
             self.last_step_time_millis = current_time
+            if step_obj.output:
+                step_obj.output.target_visible = \
```
Comment (brianpippin) [inline machine_common_sense/history_writer.py:169]: @deanwetherby  - do you mean just making this step_obj.target_visible here so we just add a value to the dictionary here and do nothing anywhere else?  I think that would give us the output we wanted and not have to worry about the metadata objects, right?

Comment (deanwetherby) [reply machine_common_sense/history_writer.py:169]: Yes. You have the gist of it

Comment (deanwetherby) [reply machine_common_sense/history_writer.py:169]: I think `step_obj` is a SceneHistory class though. We may have to add the attribute to that class but we won't run into serialization issues or confusion over what the field means in the metadata classes.

Comment (deanwetherby) [reply machine_common_sense/history_writer.py:169]: That particular method argument has step_obj mistyped as Dict.

**Review by deanwetherby — APPROVED** [review]
Feels weird approving my own branch merge into this PR but it is what it is.

**Review by brianpippin — APPROVED** [review]

**Pull request merged**